torch/csrc/jit/passes/freeze_module.cpp

```diff
@@ -567,12 +548,27 @@ class AttributePropagator {
         }
         if (n->kind() == prim::GetAttr) {
           auto& name = n->s(attr::name);
-          auto mptr =
-              getModulePtrForGetAttrNode(n->input(0)->node(), graph, module_);
-          auto module = Module(mptr);
-          if (module.type() == n->inputs()[0]->type() && module.hasattr(name)) {
-            auto attr = module.attr(name);
-            insertMutableAttr(name, attr, mptr);
+          // For now, use all module ivalues which are the same type
+          // and could be the module that this GetAttr resolves to
+          // TODO: we could attempt to follow the GetAttr chain and 
+          // find the exact ivalue, we would have to be careful
+          // that the chain does not contain any attributes which 
+          // get written to (setAttr calls)
+          for (auto& mptr : modules) {
+            auto module = Module(mptr);
+            if (module.type() == n->inputs()[0]->type()) {
+              TORCH_INTERNAL_ASSERT(module.hasattr(name));
+              auto module = Module(mptr);
+              auto attr = module.attr(name);
+              // TODO: this could be insertReferencedAttr to be more clear,
+              // these are attributes we could not inline, which include 
+              // other reasons besides mutation (unsupported constant, 
+              // getAttr resolving to non-getAttr node, etc)
+              insertMutableAttr(name, attr, mptr);
+              if (attr.isModule()) {
+                modules.insert(attr.toModule()._ivalue());
```

Comment:
I dont quite follow why not breaking from loop helps. Does it mean that you gonna now add modules you get from 'getattr' nodes to the list and always iter over all of them so as not to not skip any of them. And then for a getattr node also match against all possible instances of the module type?

Reply:
Yes, it means for getAttr we're matching against all possible instances of the module type, instead of just the first one (first version) or the one that the gettAttr chain resolves to bc:
> we would have to be careful that the chain does not contain any attributes which get written to (setAttr calls)

Reply:
We could try to be more exact and follow getAttrs calls + make sure that none of those attributes were mutable, but that should be a follow up, we should get correctness down + make freezing not fail on models first

Reply:
Yeah that sounds good. In few instances of reported breakage of freeze_module because of this PR was generally related to this issue of not being able follow chain of getattrs. Thanks for the test case. I kind of see that now.